mlir/lib/Quantum/Transforms/ChainedSelfInversePatterns.cpp

```diff
@@ -41,34 +43,176 @@ struct ChainedHadamardOpRewritePattern : public mlir::OpRewritePattern<CustomOp>
         LLVM_DEBUG(dbgs() << "Simplifying the following operation:\n" << op << "\n");
 
         StringRef OpGateName = op.getGateName();
-        if (!HermitianOps.contains(OpGateName))
+        if (!HermitianOps.contains(OpGateName)) {
             return failure();
+        }
 
         ValueRange InQubits = op.getInQubits();
         auto ParentOp = dyn_cast_or_null<CustomOp>(InQubits[0].getDefiningOp());
-        if (!ParentOp || ParentOp.getGateName() != OpGateName)
+        if (!ParentOp || ParentOp.getGateName() != OpGateName) {
             return failure();
+        }
 
         ValueRange ParentOutQubits = ParentOp.getOutQubits();
         // Check if the input qubits to the current operation match the output qubits of the parent.
         for (const auto &[Idx, Qubit] : llvm::enumerate(InQubits)) {
-            if (Qubit.getDefiningOp<CustomOp>() != ParentOp || Qubit != ParentOutQubits[Idx])
+            if (Qubit.getDefiningOp<CustomOp>() != ParentOp || Qubit != ParentOutQubits[Idx]) {
                 return failure();
+            }
         }
         ValueRange simplifiedVal = ParentOp.getInQubits();
         rewriter.replaceOp(op, simplifiedVal);
         return success();
     }
 };
 
+template <typename OpType>
```

Comment:
It might be better to use the existing gate op interfaces if possible

https://github.com/PennyLaneAI/catalyst/blob/main/mlir/include/Quantum/IR/QuantumInterfaces.td